io/transmat/gs/gs_token.go

```diff
@@ -0,0 +1,77 @@
+package gs
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/storage/v1"
+)
+
+const EnvGsAccessToken = "GS_ACCESS_TOKEN"
+const EnvGsAccountFile = "GS_SERVICE_ACCOUNT_FILE"
+
+func GetAccessToken() (*oauth2.Token, error) {
+	var err error
+	var secrets []byte
+	token := getAccessTokenFromEnv()
+	if token != nil {
+		return token, nil
+	}
+	secrets, err = getServiceAccountSecretsFromEnv()
+	if err == nil {
+		token, err = createAccessTokenFromSecrets(secrets, []string{storage.DevstorageReadWriteScope})
+	}
+	return token, err
+}
+
+func getAccessTokenFromEnv() *oauth2.Token {
+	accessToken := os.Getenv(EnvGsAccessToken)
+	if accessToken != "" {
+		return &oauth2.Token{AccessToken: accessToken}
```

Comment:
This would be nice, but [RFC6749 Section 1.4: Access tokens](https://tools.ietf.org/html/rfc6749#section-1.4) says that access tokens can be opaque to the client and the format is based on the server's requirements. We could implement something based on assumptions about Google's OAuth2 implementation, but ultimately it boils down to "try and see" after eliminating zero values. Anything else will probably be fragile and unsatisfying.